src/Groups/schur_index.jl

```diff
@@ -0,0 +1,520 @@
+# AbstractAlgebra.set_verbosity_level(:SchurIndices, 1)
+
+"""
+    yamada_example(p::Int, d::Int)
+
+Return `G, chi` where `G` is a pc group of order `p*(p-1)*d`
+and `chi` is a faithful irreducible character of degree `p-1` of `G`
+that has Schur index `d`.
+
+`p` must be a prime integer, and `d` must divide `p-1`.
+
+This example is described in [Yam78](@cite).
+
+# Examples
+```jldoctest
+julia> G, chi = Oscar.yamada_example(7, 3);
+
+julia> order(G)
+126
+
+julia> schur_index(chi)
+3
+```
+"""
+function yamada_example(p::Int, d::Int)
+  @req is_odd(p) && is_prime(p) "p must be an odd prime integer"
+  @req mod(p-1, d) == 0 "d must divide p-1"
+
+  r = GAP.Globals.PrimitiveRootMod(p)
```

Comment:
We have that functionality on Hecke, somewhere. I think (independent of this PR) we need to make it available

Reply:
Yes, there is `Hecke.gen_mod_pk`.